airbyte-integrations/bases/bases-destination-jdbc/src/main/java/io/airbyte/integrations/destination/staging/AsyncFlush.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.staging;
+
+import io.airbyte.db.jdbc.JdbcDatabase;
+import io.airbyte.integrations.destination.jdbc.WriteConfig;
+import io.airbyte.integrations.destination.record_buffer.FileBuffer;
+import io.airbyte.integrations.destination.s3.csv.CsvSerializedBuffer;
+import io.airbyte.integrations.destination.s3.csv.StagingDatabaseCsvSheetGenerator;
+import io.airbyte.integrations.destination_async.DestinationFlushFunction;
+import io.airbyte.protocol.models.Jsons;
+import io.airbyte.protocol.models.v0.AirbyteMessage;
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteCatalog;
+import io.airbyte.protocol.models.v0.StreamDescriptor;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FileUtils;
+
+/**
+ * Async flushing logic. Flushing async prevents backpressure and is the superior flushing strategy.
+ */
+@Slf4j
+class AsyncFlush implements DestinationFlushFunction {
+
+  private final Map<StreamDescriptor, WriteConfig> streamDescToWriteConfig;
+  private final StagingOperations stagingOperations;
+  private final JdbcDatabase database;
+  private final ConfiguredAirbyteCatalog catalog;
+
+  public AsyncFlush(final Map<StreamDescriptor, WriteConfig> streamDescToWriteConfig,
+                    final StagingOperations stagingOperations,
+                    final JdbcDatabase database,
+                    final ConfiguredAirbyteCatalog catalog) {
+    this.streamDescToWriteConfig = streamDescToWriteConfig;
+    this.stagingOperations = stagingOperations;
+    this.database = database;
+    this.catalog = catalog;
+  }
+
+  // todo(davin): exceptions are too broad.
+  @Override
+  public void flush(final StreamDescriptor decs, final Stream<AirbyteMessage> stream) throws Exception {
+    // write this to a file - serilizable buffer?
+    // where do we create all the write configs?
+    log.info("Starting staging flush..");
+    CsvSerializedBuffer writer = null;
```

Comment:
is this CSV general or snowflake specific?

Reply:
good point. this might be snowflake specific. i'll follow up. however, shouldn't block you from moving ahead now.

leaving comment for myself

Reply:
The `CsvSerializedBuffer` is also used for Redshift